### ISSUE TITLE
`swagger-ui.html` is not found - HTTP404

### ISSUE THREAD

**Issue by vilmosnagy** [springdoc/springdoc-openapi#402]
TLDR: this is the duplicate of the issue #361, but:
 - the solution there (start the app differently from IntelliJ) did not solve my issue
 - I was able to create an MCVE

**Describe the bug**
I do run `spring-boot` `2.2.2-RELEASE` with `springdoc-openapi-webflux-ui` `1.2.29`.

The following requests give me the following statuses:

```
[vilmosnagy@vnagy-dell hkir-trip-planner]$ curl -sD - http://localhost:8080/v3/api-docs/swagger-config  && echo 
HTTP/1.1 200 OK
Content-Type: application/json
Content-Length: 119

{"configUrl":"/v3/api-docs/swagger-config","oauth2RedirectUrl":"/swagger-ui/oauth2-redirect.html","url":"/v3/api-docs"}
[vilmosnagy@vnagy-dell hkir-trip-planner]$ curl -sD - http://localhost:8080/v3/api-docs   && echo 
HTTP/1.1 200 OK
Content-Type: application/json;charset=UTF-8
Content-Length: 719

{"openapi":"3.0.1","info":{"title":"OpenAPI definition","version":"v0"},"servers":[{"url":"http://localhost:8080","description":"Generated server url"}],"paths":{"/v1/test":{"post":{"tags":["example-controller"],"operationId":"getPlanForParameter","requestBody":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/ExampleRequest"}}}},"responses":{"200":{"description":"default response","content":{"application/json":{"schema":{"$ref":"#/components/schemas/ExampleResponse"}}}}}}}},"components":{"schemas":{"ExampleRequest":{"type":"object","properties":{"id":{"type":"string"}}},"ExampleResponse":{"type":"object","properties":{"id":{"type":"string"},"count":{"type":"integer","format":"int32"}}}}}}
[vilmosnagy@vnagy-dell hkir-trip-planner]$ curl -sD - http://localhost:8080/swagger-ui.html  && echo 
HTTP/1.1 307 Temporary Redirect
Location: /webjars/swagger-ui/index.html?configUrl=/v3/api-docs/swagger-config
content-length: 0


[vilmosnagy@vnagy-dell hkir-trip-planner]$ curl -sD - http://localhost:8080/webjars/swagger-ui/index.html?configUrl=/v3/api-docs/swagger-config  && echo 
HTTP/1.1 404 Not Found
Content-Type: application/json
Content-Length: 155

{"timestamp":1580836491212,"path":"/webjars/swagger-ui/index.html","status":404,"error":"Not Found","message":"No matching handler","requestId":"ed2cc66d"}
[vilmosnagy@vnagy-dell hkir-trip-planner]$ 
```

(TLDR: it's the same as in the mentioned issue, the `swagger-ui.html` redirects me somewhere, and that page does not load)

**To Reproduce**
 - Clone this repository: https://github.com/vilmosnagy/springdoc-webflux-404
 - Start the app with `mvn spring-boot:run` in the command line
 - try to load the Swagger-UI at http://localhost:8080/swagger-ui.html

**Expected behavior**
- I shall see the swagger IO at that location

**Screenshots**
Nothing extraordinary, the standard Spirng-Boot error page.
![image](https://user-images.githubusercontent.com/6528869/73769240-a0f1ba00-477a-11ea-9b9f-48e324dd0256.png)

**Additional context**
My problem is the same as in #361, but I was able to create an MVCE.

**Comment by bnasslahsen** [comment]
Hi @vilmosnagy, 

As stated in the Spring Boot reference documentation, when you use `@EnableWebFlux`, you tell Spring Boot that you wish to take full control over the WebFlux configuration and disable all auto-configuration for this (including static resources):
 - https://docs.spring.io/spring-boot/docs/current/reference/html/spring-boot-features.html#boot-features-webflux-auto-configuration

 You have two solutions:
 - Remove `@EnableWebFlux`
 - If you really want take control of Bean creation and you want absolutely to use `@EnableWebFlux`, then you need to add an implementation of `WebFluxConfigurer`. 
 Here is a sample code:

```java
@Configuration
public class WebConfig  implements WebFluxConfigurer {
	@Override
	public void addResourceHandlers(ResourceHandlerRegistry registry) {
		registry.addResourceHandler("/webjars/**")
				.addResourceLocations(CLASSPATH_RESOURCE_LOCATION+"/webjars/")
				.resourceChain(true)
				.addResolver(new WebJarsResourceResolver());
	}
}
```

**Comment by vilmosnagy** [comment]
Thanks for the quick reply.

**Comment by kanai0618** [comment]
Hi @bnasslahsen , **i used** 
@Configuration
public class WebConfig  implements WebFluxConfigurer {
	@Override
	public void addResourceHandlers(ResourceHandlerRegistry registry) {
		registry.addResourceHandler("/webjars/**")
				.addResourceLocations(CLASSPATH_RESOURCE_LOCATION+"/webjars/")
				.resourceChain(true)
				.addResolver(new WebJarsResourceResolver());
	}
}
still i am getting 404. and in my spring boot , in pattern **o.s.c.g.h.p.RoutePredicateFactory        : Pattern**, it doesnot find any swagger path. can you help ?

**Comment by bnasslahsen** [comment]
@kanai0618, 

 I don't know the context of your question, which is not clear and the code is not even formatted.
Unfortunately, i cannot help you without any relevant information.


  Please follow the contribution [guidelines](https://github.com/springdoc/springdoc-openapi/blob/master/CONTRIBUTING.adoc). 
 You have guidance and [demos](https://github.com/springdoc/springdoc-openapi-demos) for spring-weblfux.
 And if you are not able to make it, please help us speed up problem diagnosis by providing as much information as described in the guidelines.

**Comment by kanai0618** [comment]
Hi @bnasslahsen , Sorry about the formatting issue.

So, in my project i am using webflux and spring cloud gateway. and i have included 
```
<dependency>
            <groupId>org.springdoc</groupId>
            <artifactId>springdoc-openapi-webflux-ui</artifactId>
            <version>1.5.2</version>
        </dependency>
```
and when i run the swagger endpoint, it gives 404. and in console i can see, ```No RouteDefinition found for```, as no pattern of swagger-ui endpoint is finding.

And i also tried with including this
```@Configuration
public class WebfluxConfig implements WebFluxConfigurer {

    @Override
    public void addResourceHandlers(ResourceHandlerRegistry registry) {
        registry.addResourceHandler("*/swagger-ui.html")
                .addResourceLocations("classpath:/META-INF/resources/");
        registry.addResourceHandler("*/webjars/*")
                .addResourceLocations("classpath:/META-INF/resources/webjars/");
    }
}```
still same issue, 
Hope its clear now, if you can help with ?

**Comment by kanai0618** [comment]
Hi @bnasslahsen , sorry for the typing issue.

So i am using spring cloud gateway in my code. and i have included 
```
  <dependency>
            <groupId>org.springdoc</groupId>
            <artifactId>springdoc-openapi-webflux-ui</artifactId>
            <version>1.5.2</version>
        </dependency>
```
and i am able to load this 
```
http://localhost:8080/webjars/swagger-ui/index.html - ( by default https://petstore.swagger.io/v2/swagger.json)
```
but when i actually try to load my swagger endpoint (v3/api-docs), it gives me error.  ( http://localhost:8080/v3/api-docs) 404. not found.
hope you understand the issue.

**Comment by bnasslahsen** [comment]
@kanai0618,

 You have [here](https://github.com/springdoc/springdoc-openapi-demos/tree/master/sample-springdoc-openapi-microservices) a fully working sample of spring cloud gateway, which is working.
 You can see the [Piotrminkowski Blog]( https://piotrminkowski.com/2020/02/20/microservices-api-documentation-with-springdoc-openapi/) for  detailed  instructions

 And if you are still struggling, please make sure you Provide a [ Minimal, Reproducible Example - with HelloController that reproduces the problem](https://stackoverflow.com/help/minimal-reproducible-example)
